src/Symfony/Component/Yaml/Inline.php

```diff
@@ -320,7 +320,11 @@ private static function dumpArray($value, $flags)
      */
     public static function parseScalar($scalar, $flags = 0, $delimiters = null, &$i = 0, $evaluate = true, $references = [], $legacyOmittedKeySupport = false)
     {
-        if (\in_array($scalar[$i], ['"', "'"])) {
+        if ('' === $scalar = (string) $scalar) {
```

Comment:
This cast can be removed when merged in upper branches since the parameter is typed, so when false is passed it is converted to ''.

Reply:
The docblock states that `$scalar` is a string. Can we detect the cases where we pass something else here and terminate there instead?

Reply:
What I did first was to cast to string before calling this method, to stay compliant with the method contract, but see https://github.com/symfony/symfony/pull/35208#discussion_r363035213.

Reply:
Why do we have to call the method at all if the scalar is not a string?